contracts/SpokePool.sol

```diff
@@ -667,6 +675,60 @@ abstract contract SpokePool is
         _emitFillRelay(relayExecution, fillAmountPreFees);
     }
 
+    /**
+     * @notice Caller signals to the system that they want a refund on this chain, which they set as the
+     * `repaymentChainId` on the original fillRelay() call on the `destinationChainId`. An observer should be
+     * be able to 1-to-1 match the emitted RequestRefund event with the FilledRelay event on the `destinationChainId`.
+     * @dev This function could be used to artificially inflate the `fillCounter`, allowing the caller to "frontrun"
+     * and cancel pending fills in the mempool. This would in the worst case censor fills at the cost of the caller's
+     * gas costs. We don't view this as a major issue as the fill can be resubmitted and obtain the same incentive,
+     * since incentives are based on validated refunds and would ignore these censoring attempts. This is no
+     * different from calling `fillRelay` and setting msg.sender = recipient.
+     * @dev Caller needs to pass in `fillBlock` that the FilledRelay event was emitted on the `destinationChainId`.
+     * This is to prevent someone from requesting a refund before a corresponding fill has been mined.
+     * @param relayer Relayer on FilledRelay event on destinationChain we want to match with this Refund.
+     * This should be account receiving any refund stemming from this event.
+     * @param destinationToken This chain's destination token equivalent for original deposit destination token.
+     * @param amount Original deposit amount.
+     * @param originChainId Original origin chain ID.
+     * @param realizedLpFeePct Original realized LP fee %.
+     * @param depositId Original deposit ID.
+     * @param maxCount Max count to protect the refund recipient from frontrunning.
+     * @param fillBlock Block number of FilledRelay event on destinationChain we want to match with this Refund.
+     */
+    function requestRefund(
```

Comment:
Is there any concern about duplicates? Will duplicates break anything?

Reply:
Duplicates shouldn't break anything as long as bots have a determinsitic way to choose one, so say, pick the earliest match?

Reply:
I think the duplicate count should be emitted in the event so the bot can be absolutely sure that it doesn't double refund even if missing an event.

Reply:
I have added duplicate counting to this PR.